### PR TITLE
[AIRFLOW-3821] Add replicas logic to GCP SQL example DAG

### PR DESCRIPTION
**Pull request by antonimaciej**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: docs/howto/operator.rst [hunk 1]
```diff
@@ -1170,6 +1172,8 @@ it will be retrieved from the GCP connection used. Both variants are shown:
     :start-after: [START howto_operator_cloudsql_delete]
     :end-before: [END howto_operator_cloudsql_delete]
 
+Note: If the instance has read or failover replicas you need to delete them first.
```
Comment (potiuk) [inline docs/howto/operator.rst]: I'd say "you need to delete them before you delete the primary instance.".

Comment (antonimaciej) [reply docs/howto/operator.rst]: Sure, I will change it.

Comment (antonimaciej) [reply docs/howto/operator.rst]: Done.

File: docs/howto/operator.rst [hunk 2]
```diff
@@ -1386,13 +1392,30 @@ Some arguments in the example DAG are taken from OS environment variables:
...
+Example body defining read replica for the instance above:
+
+.. literalinclude:: ../../airflow/contrib/example_dags/example_gcp_sql.py
+    :language: python
+    :start-after: [START howto_operator_cloudsql_create_replica]
+    :end-before: [END howto_operator_cloudsql_create_replica]
+
+Note: Failover replicas are created together with the instance in a single task.
+Read replicas need to be created in separate tasks.
+
```
Comment (sprzedwojski) [inline docs/howto/operator.rst]: I'd add example usages of the create / delete operator to create and delete read / failover replicas from the example DAG somewhere here.

Comment (antonimaciej) [reply docs/howto/operator.rst]: Example of creating failover and read replicas is documented few lines above that notes :)
Those notes refer to those examples.
However I will add deletion examples to documentation (in the CloudSqlInstanceDeleteOperator section)

Comment (antonimaciej) [reply docs/howto/operator.rst]: Examples added

Comment (sprzedwojski) [reply docs/howto/operator.rst]: Ok for the delete example. However, for the create, I'd still add a separate code example here for creating the read replica (under "Using the operator").

File: docs/howto/operator.rst [hunk 3]
```diff
@@ -1386,13 +1399,30 @@ Some arguments in the example DAG are taken from OS environment variables:
...
 Using the operator
 """"""""""""""""""
```
Comment (sprzedwojski) [inline docs/howto/operator.rst]: Ok for the delete example. However, for the create, I'd still add a separate code example here for creating the read replica (under "Using the operator").

**Pull request closed without merging**